### PR TITLE
Update documentation.

### PR DESCRIPTION
**Pull request by notlee**
- Remove TROUBLESHOOT.md, these instrutions apply to any globally
install node cli and is probably not very helpful to send users
there first.
- Move migration guides to `MIGRATION.md`
- Add a "Developing Existing Components Locally" and "Creating A
New Component" section, for users who would rather follow steps
than figure out all the api options (these sections could be
expanded in the future, or link to a tutorial on origami.ft.com)
- Replace references to modules with components.
- Remove references of projects other than components.

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+
+
+### Migrating from v9 to v10
```
Comment (chee) [inline MIGRATION.md:3]:
```suggestion
### Migrating from v9 to v10
```

File: README.md [hunk 2]
```diff
@@ -40,116 +73,57 @@ If you have any issues with OBT, please check out [troubleshooting guide](https:
 		--brand=<brand-name>       Build SCSS for a given brand. E.G. --brand=internal to build the component for the internal brand.
 		--debug                    Keep the test runner open to enable debugging in any browser.
 
-### Developing components locally
-
-Build and browse the demos (typically: <http://localhost:8080/demos/local/>),
-automatically re-build the component's demos and assets every time a file changes:
-
-	`obt dev`
+### Commands
```
Comment (chee) [inline README.md:76]: is this meant to be a subsection under usage? it looks from the ToC that Commands is a h2-level section

Comment (notlee) [reply README.md:76]: good point, I'll make commands and api reference h2

File: README.md [hunk 3]
```diff
@@ -1,34 +1,68 @@
...
+
+Also run the components tests:
+
+`obt test`
+
+To debug JavaScript test errors with immediate feedback and breakpoints, pass the `debug` flag. This will open the tests in a browser window with logs in the browser console:
+
+`obt test --debug`
+
+Further, JavaScript tests may be run in BrowserStack automatically with the `--browserstack` flag. See the [test command](#test-or-t) for more details.
```
Comment (chee) [inline README.md:50]: this is a much nicer interface

**Pull request merged**